spec/values/numbers/degenerate.hrx

```diff
@@ -0,0 +1,195 @@
+<===> nan/single_unit/input.scss
+@use 'sass:math';
+a {b: math.div(0px, 0)}
+
+<===> nan/single_unit/output.css
+a {
+  b: calc(NaN * 1px);
+}
+
+<===>
+================================================================================
+<===> error/infinity/multiple_numerator_units/input.scss
+@use 'sass:math';
+a {b: math.div(1px * 1em, 0)}
+
+<===> error/infinity/multiple_numerator_units/error
+Error: calc(Infinitypx*em) isn't a valid CSS value.
+  ,
+2 | a {b: math.div(1px * 1em, 0)}
+  |       ^^^^^^^^^^^^^^^^^^^^^^
+  '
+  input.scss 2:7  root stylesheet
+
+<===>
+================================================================================
+<===> error/infinity/denominator_unit/input.scss
+@use 'sass:math';
+a {b: math.div(1, 0px)}
+
+<===> error/infinity/denominator_unit/error
+Error: calc(Infinitypx^-1) isn't a valid CSS value.
+  ,
+2 | a {b: math.div(1, 0px)}
+  |       ^^^^^^^^^^^^^^^^
+  '
+  input.scss 2:7  root stylesheet
+
+<===>
+================================================================================
+<===> error/infinity/numerator_and_denominator_unit/input.scss
+@use 'sass:math';
+a {b: math.div(1px, 0em)}
+
+<===> error/infinity/numerator_and_denominator_unit/error
+Error: calc(Infinitypx/em) isn't a valid CSS value.
+  ,
+2 | a {b: math.div(1px, 0em)}
+  |       ^^^^^^^^^^^^^^^^^^
+  '
+  input.scss 2:7  root stylesheet
+
+<===>
+================================================================================
+<===> error/minus_infinity/multiple_numerator_units/input.scss
+@use 'sass:math';
+a {b: math.div(-1px * 1em, 0)}
+
+<===> error/minus_infinity/multiple_numerator_units/error
+Error: calc(-Infinitypx*em) isn't a valid CSS value.
+  ,
+2 | a {b: math.div(-1px * 1em, 0)}
+  |       ^^^^^^^^^^^^^^^^^^^^^^^
+  '
+  input.scss 2:7  root stylesheet
+
+<===>
+================================================================================
+<===> error/minus_infinity/denominator_unit/input.scss
+@use 'sass:math';
+a {b: math.div(-1, 0px)}
+
+<===> error/minus_infinity/denominator_unit/error
+Error: calc(-Infinitypx^-1) isn't a valid CSS value.
+  ,
+2 | a {b: math.div(-1, 0px)}
+  |       ^^^^^^^^^^^^^^^^^
+  '
+  input.scss 2:7  root stylesheet
+
+<===>
+================================================================================
+<===> error/minus_infinity/numerator_and_denominator_unit/input.scss
+@use 'sass:math';
+a {b: math.div(-1px, 0em)}
+
+<===> error/minus_infinity/numerator_and_denominator_unit/error
+Error: calc(-Infinitypx/em) isn't a valid CSS value.
+  ,
+2 | a {b: math.div(-1px, 0em)}
+  |       ^^^^^^^^^^^^^^^^^^^
+  '
+  input.scss 2:7  root stylesheet
+
+<===>
+================================================================================
+<===> error/nan/multiple_numerator_units/input.scss
+@use 'sass:math';
+a {b: math.div(0px * 0em, 0)}
+
+<===> error/nan/multiple_numerator_units/error
+Error: calc(NaNpx*em) isn't a valid CSS value.
```

Comment:
Nit: I think that for `NaN` and `infinity`, having the units adjacent to the value makes the error difficult to read for me. I wonder if it would be possible to have a space in between them only for the cases of `NaN` and `infitnity`. 🤔

Reply:
I think a better solution here would be to have a more principled behavior for representing multi-unit values in general using calc, but that's outside the scope of the current change.